docs/root/configuration/listeners/listener_filters/tls_inspector.rst

```diff
@@ -18,6 +18,9 @@ of a :ref:`FilterChainMatch <envoy_v3_api_msg_config.listener.v3.FilterChainMatc
 * This filter should be configured with the type URL ``type.googleapis.com/envoy.extensions.filters.listener.tls_inspector.v3.TlsInspector``.
 * :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.listener.tls_inspector.v3.TlsInspector>`
 
+If the filter fails to detect TLS it will populate dynamic metadata indicating the reason (eg. ``ClientHello`` too
```

Comment:
Sorry for a drive-by comment. Can you put the metadata key here? You can also add this filter to https://www.envoyproxy.io/docs/envoy/latest/configuration/advanced/well_known_dynamic_metadata to ease discovery.